pennylane/interfaces/batch/jax.py

```
@@ -0,0 +1,183 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for adding the JAX interface
+to a PennyLane Device class.
+"""
+# pylint: disable=too-many-arguments
+import jax
+from jax.experimental import host_callback
+import jax.numpy as jnp
+import numpy as np
+import inspect
+
+import pennylane as qml
+
+dtype = jnp.float32
+
+
+def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_diff=2):

```

I guess we should raise an error if `max_diff>1` 😆